gdeep/trainer/regularizer.py

```diff
@@ -0,0 +1,432 @@
+from gdeep.trainer import Trainer
+import os
+import copy
+import time
+from functools import wraps
+import warnings
+from typing import Tuple, Optional, Callable, Any, Dict, List, Type, Union
+import torch.nn.functional as f
+from torch.optim import Optimizer
+import torch
+from optuna.trial._base import BaseTrial
+import numpy as np
+from tqdm import tqdm
+from sklearn.model_selection._split import BaseCrossValidator
+from sklearn.model_selection import KFold, train_test_split
+from torch.utils.data.sampler import SubsetRandomSampler
+from torch.utils.data import DataLoader
+import optuna
+from datetime import datetime
+from torch.utils.tensorboard.writer import SummaryWriter
+from torch.optim.lr_scheduler import _LRScheduler  # noqa
+from ..utility.optimization import MissingClosureError
+from gdeep.models import ModelExtractor
+from gdeep.utility import _inner_refactor_scalars
+from gdeep.utility import DEVICE
+from .metrics import accuracy
+from gdeep.utility.custom_types import Tensor
+from abc import ABCMeta, abstractmethod
+import torch.nn.functional as F
+from torch.utils.data import TensorDataset, DataLoader
+from sklearn.neighbors import KDTree
+from gudhi import SimplexTree
+import numpy as np
+from gtda.homology import VietorisRipsPersistence
+import itertools
+import matplotlib.pyplot as plt
+from gtda.plotting import plot_point_cloud
+from gtda.local_homology import KNeighborsLocalVietorisRips, RadiusLocalVietorisRips
+from gtda.diagrams import PersistenceEntropy
+from sklearn.preprocessing import FunctionTransformer
+from gtda.pipeline import make_pipeline
+
+
+def local_homology_preprocess(
+    X: Tensor,
+    n_neighbors: Tuple[int, int],
+    homology_dimensions: Tuple[int, ...],
+    dim: int,
+) -> np.ndarray:
+    """
+    A helper file for topological regularizer
+    Computes the local homology, used for the linear surrogate function
+    Args:
+        X: The point cloud on which the surrogate function is evaluated
```

Comment:
please put the description one line below, so that it appears nicely on the sphinx html.

like so:
```
Args:
    X:
        The point cloud ....
    param2:
        The description of ....
```

Reply:
done